include/magma/vk/buffers/BufferManager.hpp

```diff
@@ -0,0 +1,40 @@
+#include "magma/vk/buffers/BufferManager.h"
+template<class T>
+Buffer& BufferManager::createBufferWithData(const std::string &name, const std::vector<T> &data, 
+        VkBufferUsageFlags usage, VkMemoryPropertyFlags properties) {
+    if (bufferExists(name)) {
```

Comment:
This check comes up to 3 times for one argument with all these overloads

Reply:
Done